src/ucp/tag/tag_recv.c

```diff
@@ -100,6 +100,12 @@ static UCS_F_ALWAYS_INLINE ucs_status_ptr_t ucp_tag_recv_common(
                                                  recv_len, 1, param);
         ucp_recv_desc_release(rdesc);
 
+        if ((param->op_attr_mask & UCP_OP_ATTR_FIELD_RECV_INFO) &&
+            !(param->op_attr_mask & UCP_OP_ATTR_FLAG_NO_IMM_CMPL) &&
```

Comment:
ucp_request_imm_cmpl_param already checks UCP_OP_ATTR_FLAG_NO_IMM_CMPL flag, can we unite with it? to avoid extra branch

Reply:
split macro and using first half